Mathlib/GroupTheory/MonoidLocalization/Basic.lean

```diff
@@ -1330,3 +1330,38 @@ protected def localizationMap : S.LocalizationMap R[S⁻¹] := Localization.mono
 protected noncomputable def equivMonoidLocalization : Localization S ≃* R[S⁻¹] := MulEquiv.refl _
 
 end OreLocalization
+
+/-! ### Grothendieck group -/
+
+open Localization
+
+variable {M : Type*} [CommMonoid M]
+
+variable (M) in
+/-- The Grothendieck group of a monoid `M` is the localization at its top submonoid. -/
+@[to_additive
+"The Grothendieck group of an additive monoid `M` is the localization at its top submonoid."]
+abbrev GrothendieckGroup : Type _ := Localization (⊤ : Submonoid M)
```

Comment:
I'm not sure if this should live in the root namespace. The first thing I think of when I hear "grothendieck group" is the category theoretical construction.